### PR TITLE
Update `githelp` to advise when `sl clean --dirs` is necessary

### LINKED ISSUE #564: `sl clean --all .`, doesn't.
Despite this claim:

```shell
> sl githelp -- git clean -dfx .
sl clean --all .
```

Running `sl clean --all .` doesn't remove untracked-and-ignored stuff like my `node_modules/` folder.

---
Sapling 0.2.20230228-144002-h9440b05e

### PR DESCRIPTION
**Pull request by steveluscher**
Whenever there's a path or the `-d` flag, `git clean` implies that cleaning should be done recursively, into directories, removing empty ones as it goes.

The equivalent in Sapling is `sl clean --dirs`.

Closes #564.

### CODE REVIEW

**Comment by steveluscher** [comment]
I haven't actually run the tests yet, because I was too lazy to figure out how. Kind of hoping you'll just mash the ‘run workflows’ button on this PR. :)

**Comment by vegerot** [comment]
It's really easy to run tests (but undocumented >.>)

```sh
python run-tests.py test-fb-ext-githelp.t
```

**Comment by steveluscher** [comment]
```shell
~/src/sapling/eden/scm/tests$ python3 run-tests.py test-fb-ext-githelp.t
usage: run-tests.py [options] [tests]
run-tests.py: error: --local specified, but sl or hg not found in '~/src/sapling/eden/scm' or not executable
```

I couldn't build, and didn't want to solve the 2^53 errors that the build process spit off. If you can mash the ‘run workflows’ button that would be awesome.

**Comment by steveluscher** [comment]
OK, I got the tests running but I don't really know what I'm supposed to do here. The tests presume Mercurial, and the runs replace everything with `sl` so they fail. Halp?

```diff
   $ hg githelp -- git checkout --force
-  hg revert --all
+  sl revert --all
 
 githelp for grep with pattern and path
   $ hg githelp -- grep shrubbery flib/intern/
-  hg grep shrubbery flib/intern/
+  sl grep shrubbery flib/intern/
 
 githelp for reset, checking ~ in git becomes ~1 in mercurial
   $ hg githelp -- reset HEAD~
-  Mercurial has no strict equivalent to `git reset`.
-  If you want to remove a commit, use `hg hide -r HASH`.
-  If you want to move a bookmark, use `hg book -r HASH NAME`.
-  If you want to undo a commit, use `hg uncommit.
-  If you want to undo an amend, use `hg unamend.
+  Sapling has no strict equivalent to `git reset`.
+  If you want to remove a commit, use `sl hide -r HASH`.
+  If you want to move a bookmark, use `sl book -r HASH NAME`.
+  If you want to undo a commit, use `sl uncommit.
+  If you want to undo an amend, use `sl unamend.
```

**Comment by zzl0** [comment]
Thanks for your PR, taking a look

File: eden/scm/edenscm/ext/githelp.py [hunk 1]
```diff
@@ -384,8 +384,10 @@ def clean(ui, repo, *args, **kwargs) -> None:
     args, opts = parseoptions(ui, cmdoptions, args)
 
     cmd = Command("clean")
+    if opts.get("d") or len(args) > 0:
+        cmd["--dirs"] = None
```
Comment (zzl0) [inline eden/scm/edenscm/ext/githelp.py:388]: For Sapling, `--dirs` only delete empty directories, so I think need to set both `--dirs` and `--files` option.

**Review by zzl0 — APPROVED** [review]
Just a small comment, I will import it and fix it, thanks for your contribution.

**Comment by steveluscher** [comment]
Thanks! It's been a bit over a year since I've submitted a diff to Phabricator.

**Pull request closed without merging**